client/homebrew/editor/snippetbar/snippetbar.jsx

```diff
@@ -58,14 +62,21 @@ const Snippetbar = createClass({
 	},
 
 	componentDidUpdate : async function(prevProps) {
+		const update = {};
+		let newData = false;
```

Comment:
There's no need to try to bundle the setState into one update object like this. React does this automatically. Just use `this.setState({snippets : snippets})` and then `this.setState((prevState)=>{thistoryExists : !prevState.historyExists});` or whatever so the two update steps don't have to be intertwined.

Reply:
I've disentangled them now; it all seems happy. Pushing the commit shortly.